collector/src/main/java/com/bitmovin/analytics/data/HttpBackend.java

```diff
@@ -18,28 +26,61 @@ public HttpBackend(CollectorConfig config, Context context) {
         analyticsBackendUrl = Uri.parse(config.getBackendUrl()).buildUpon().appendEncodedPath("analytics").build().toString();
         adsAnalyticsBackendUrl = Uri.parse(config.getBackendUrl()).buildUpon().appendEncodedPath("analytics/a").build().toString();
         Log.d(TAG, String.format("Initialized Analytics HTTP Backend with %s", analyticsBackendUrl));
-        this.httpClient = new HttpClient(context);
+        this.httpClient = new HttpClient(context, new ClientFactory().createClient(config));
     }
 
     @Override
     public void send(EventData eventData) {
+        this.send(eventData, null);
+    }
+
+    @Override
+    public void sendAd(AdEventData eventData) {
+        this.sendAd(eventData, null);
+    }
+
+    @Override
+    public void send(EventData eventData, OnFailureCallback callback) {
         Log.d(TAG, String.format("Sending sample: %s (state: %s, videoId: %s, startupTime: %d, videoStartupTime: %d, buffered: %d, audioLanguage: %s)",
                 eventData.getImpressionId(),
                 eventData.getVideoId(),
                 eventData.getState(),
                 eventData.getStartupTime(),
                 eventData.getVideoStartupTime(),
-                eventData.getBuffered(), 
+                eventData.getBuffered(),
                 eventData.getAudioLanguage()));
-        this.httpClient.post(analyticsBackendUrl, DataSerializer.serialize(eventData), null);
+        this.httpClient.post(analyticsBackendUrl, DataSerializer.serialize(eventData), new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                if(callback != null) {
+                    callback.onFailure(e, () -> { call.cancel(); return  null;});
+                }
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) {
+            }
+        });
     }
 
     @Override
-    public void sendAd(AdEventData eventData) {
+    public void sendAd(AdEventData eventData, OnFailureCallback callback) {
         Log.d(TAG, String.format("Sending ad sample: %s (videoImpressionId: %s, adImpressionId: %s)",
                 eventData.getAdImpressionId(),
                 eventData.getVideoImpressionId(),
                 eventData.getAdImpressionId()));
-        this.httpClient.post(adsAnalyticsBackendUrl, DataSerializer.serialize(eventData), null);
+        this.httpClient.post(adsAnalyticsBackendUrl, DataSerializer.serialize(eventData), new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                if(callback !=null) {
```

Comment:
invert and formatting `!=null`

Can you run the formatter (either inside intelliJ or ktlint / coveo?)

Reply:
fixed in https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/132/commits/8ed6142f98230909ac19f34acadc4ad135cae400